src/wui/mapdetails.cc

```diff
@@ -201,15 +201,27 @@ void MapDetails::update(const MapData& mapdata, bool localize_mapname) {
 			egbase_.cleanup_for_load();
 			std::unique_ptr<Widelands::MapLoader> ml(
 			   egbase_.mutable_map()->get_correct_loader(mapdata.filename));
-			if (ml.get() && 0 == ml->load_map_for_render(egbase_)) {
-				minimap_cache_[last_map_] = draw_minimap(
-				   egbase_, nullptr, Rectf(), MiniMapType::kStaticMap,
-				   MiniMapLayer::Terrain | MiniMapLayer::StartingPositions | MiniMapLayer::Owner);
-				minimap_icon_.set_icon(minimap_cache_.at(last_map_).get());
-				minimap_icon_.set_visible(true);
+			try {
+				if (ml.get() && 0 == ml->load_map_for_render(egbase_)) {
+					minimap_cache_[last_map_] = draw_minimap(
+					   egbase_, nullptr, Rectf(), MiniMapType::kStaticMap,
+					   MiniMapLayer::Terrain | MiniMapLayer::StartingPositions | MiniMapLayer::Owner);
+					minimap_icon_.set_icon(minimap_cache_.at(last_map_).get());
+					minimap_icon_.set_visible(true);
+				}
+			} catch (const Widelands::GameDataError& e) {
+				// Put error message on top for better visibility
+				description =
+				   (boost::format("%s%s") % as_content(e.what(), style_) % description).str();
+				description =
+				   (boost::format("%s%s") % as_heading(_("Game data error"), style_) % description)
+				      .str();
```

Comment:
```suggestion
				description =
				   (boost::format("%s%s%s") % as_heading(_("Game data error"), style_) % as_content(e.what(), style_) % description).str();
```

Reply:
I simply adapted the previous style. If you want to reduce boost::format calls, I should apply this also to the other formatting and always combine content and heading in one call?

Reply:
If it's consistent with the other calls then it's up to you whether you want to change this or not :)

Reply:
Then for readability I suggest to keep it.